src/lib_json/json_value.cpp

```diff
@@ -1165,6 +1166,28 @@ Value& Value::append(Value&& value) {
   return this->value_.map_->emplace(size(), std::move(value)).first->second;
 }
 
+/// \brief Insert value in array at specific index
+bool Value::insert(ArrayIndex index, const Value& newValue) {
+  JSON_ASSERT_MESSAGE(type() == nullValue || type() == arrayValue,
+                      "in Json::Value::insert(ArrayIndex index, const Value& "
+                      "newValue): requires arrayValue");
+  return insert(index, Value(newValue));
```

Comment:
If we always copy, why do we have a const& function parameter?

You can replace both of these methods with 

`Value::insert(ArrayIndex index, Value newValue)` then people can copy or move based on how they call it.